client/my-sites/domains/domain-management/transfer/transfer-to-other-site/confirmation-dialog.jsx

```diff
@@ -38,27 +69,24 @@ class TransferConfirmationDialog extends React.PureComponent {
 			},
 			{
 				action: 'confirm',
-				label: translate( 'Confirm Transfer' ),
+				label: actionLabel,
 				onClick: this.onConfirm,
 				disabled: this.props.disableDialogButtons,
 				isPrimary: true,
 			},
 		];
 
-		const targetSiteTitle = get( this.props.targetSite, 'title', '' );
-
 		return (
 			<Dialog isVisible={ this.props.isVisible } buttons={ buttons } onClose={ this.props.onClose }>
 				<h1>{ translate( 'Confirm Transfer' ) }</h1>
+				<p>{ this.getMessage() }</p>
 				<p>
-					{ translate(
-						'Do you want to transfer {{strong}}%(domainName)s{{/strong}} ' +
-							'to site {{strong}}%(targetSiteTitle)s{{/strong}}?',
-						{
-							args: { domainName, targetSiteTitle },
-							components: { strong: <strong /> },
-						}
-					) }
+					{ this.props.primaryWithPlansOnly &&
+						! this.props.isTargetSiteOnPaidPlan &&
+						translate(
+							"The target site is not a paid plan, so you'll have to pay the full price for the domain mapping subscription on renewal. " +
+								'Consider upgrading the site to a paid plan to have the mapping for free.'
+						) }
 				</p>
```

Comment:
The change to `TransferToOtherSite.isSiteEligible()` to allow transfers to sites with a free plan means that we also need to communicate that this domain can not be set as the primary domain unless the target site gets upgraded, _or_ the user has an exemption to make the domain primary. (This applies to both domains and domain mappings, AFAIK.)

Reply:
I've adjusted the copy a bit to mention that explicitly. To be honest, this whole flow is not the most user-friendly, but we have to do with what we have. At the very least, it's an action that can be "easily" reverted (the domain transferred back), if the user notices that on the target site they cannot set it as primary and that's a blocker for them. It's, of course, far from ideal, but I'd want to first have a parity between mappings and registrations and then we can try to improve PWPO handling. Or, hopefully, just remove the limitation at all ;)